src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,60 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { SelectCategory } from '../../types/SelectCategory';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+type Props = {
+  query: string;
+  setQuery: (value: string) => void;
+  setSelectCategory: (value: SelectCategory) => void;
+};
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = props => {
+  const { query, setQuery, setSelectCategory } = props;
+
+  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectCategory(event.target.value as SelectCategory);
```

Comment:
The type casting `event.target.value as SelectCategory` assumes that the value of the select options matches the `SelectCategory` enum values. Ensure that the values in the `<option>` elements (`SelectCategory.all`, `SelectCategory.active`, `SelectCategory.completed`) are correctly defined in the `SelectCategory` enum to avoid runtime errors.